src/controller/CHIPDeviceController.cpp

```diff
@@ -2170,9 +2187,43 @@ void DeviceCommissioner::ParseFabrics()
         mPairingDelegate->OnFabricCheck(info.nodeId);
     }
 
-    CommissioningDelegate::CommissioningReport report;
-    report.Set<MatchingFabricInfo>(info);
-    CommissioningStageComplete(return_err, report);
+    return return_err;
+}
+
+CHIP_ERROR DeviceCommissioner::ParseICDInfo(ReadCommissioningInfo2 & info)
+{
+    CHIP_ERROR err;
+    IcdManagement::Attributes::FeatureMap::TypeInfo::DecodableType featureMap;
+
+    err = mAttributeCache->Get<IcdManagement::Attributes::FeatureMap::TypeInfo>(kRootEndpointId, featureMap);
+    if (err == CHIP_NO_ERROR)
+    {
+        info.isIcd                  = true;
+        info.checkInProtocolSupport = !!(featureMap & to_underlying(IcdManagement::Feature::kCheckInProtocolSupport));
+    }
+    else if (err == CHIP_ERROR_KEY_NOT_FOUND)
```

Comment:
So this would be a case of a non-spec-compliant server that did not respond at all to a non-wildcard path?

Please document exactly why this case is here, because it's not obvious to readers how this situation could happen.

And perhaps this case should just lead comissioning to fail, just like an error other than UnsupportedCluster causes it to fail?